### ISSUE TITLE
Does this work with Razor views?

### ISSUE THREAD

**Issue by danieljsummers** [aspnet/Docs#6796]
This is a feature I've been wanting for quite some time. :) I've been trying to make it work by providing Razor views for use in an MVC application, but I'm having trouble getting that to work. From the description, it seems like it should work; am I reading correctly?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: d92ffa7c-fd65-7dc3-194e-6a67a8a7f176
* Version Independent ID: 2df11a37-d64e-6353-bd92-0b7729070c3e
* Content: [Reusable Razor UI in class libraries with ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/mvc/razor-pages/ui-class?view=aspnetcore-2.1&tabs=visual-studio)
* Content Source: [aspnetcore/mvc/razor-pages/ui-class.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/mvc/razor-pages/ui-class.md)
* Service: **unspecified**
* Product: **asp.net-core**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

**Comment by Rick-Anderson** [comment]
Yes. https://docs.microsoft.com/en-us/aspnet/core/tutorials/razor-pages/page?view=aspnetcore-2.1 shows it with MVC apps.

**Comment by rahamohebbi** [comment]
Similar to @danieljsummers I can't get the RCL working with View Component. I am getting an error saying that the view can't be found.

**Comment by Rick-Anderson** [comment]
@danieljsummers  oops, wrong link. Try https://docs.microsoft.com/en-us/aspnet/core/security/authentication/scaffold-identity?view=aspnetcore-2.1&tabs=visual-studio#scaffold-identity-into-an-mvc-project-without-existing-authorization

**Comment by Rick-Anderson** [comment]
@rahamohebbi  create the simplest possible project that reproduces the problem and I'll see if I can get someone to look at it. I know it works with view components though.

**Comment by danieljsummers** [comment]
I've tried several different spots, but still can't make it. I've made a [repro of it](https://github.com/danieljsummers/rcl-test) with 4 different patterns, none of which work.

(This is low priority; I'm sure it's something I'm doing wrong, and I truly do appreciate the help.)

**Comment by danieljsummers** [comment]
BTW, I'd also be willing to strip it down further (remove JS and all that) and make a similar example as the one with the pages.

**Comment by Rick-Anderson** [comment]
@pranavkm can you comment on problems with VC's in RCLs?

**Comment by Rick-Anderson** [comment]
ASP.NET Core discovers view components from all registered application parts and `ViewComponent` **views** should compile along with other views in the project just fine.

**Comment by pranavkm** [comment]
@danieljsummers - try changing the name of your secondary project. The file name that the class library `Rcl.Views` produces collides with the Views dll that the RazorSdk produces for the Rcl project.

@rahamohebbi a repro app would help.

**Comment by rahamohebbi** [comment]
@Rick-Anderson @pranavkm 
I created a brand new asp net core application and added a brand new class library (RCL). everything worked really well so started digging a little more comparing it with my main application.

The only difference i noticed was the RCL project file. copy pasted it to my existing class library and everything worked as it should. We might have some bugs there that needs to be ironed out.

Cheers

**Comment by Rick-Anderson** [comment]
@rahamohebbi  thanks for the update.

**Comment by Rick-Anderson** [comment]
@pranavkm  can you respond to @danieljsummers 

> Would this page be the right place for a warning about this? I'll do up a PR if so.

Is that worth warning about?

**Comment by danieljsummers** [comment]
While you're considering it, my thoughts were, under the "Create a class library containing Razor UI" heading:

- On the "Visual Studio" tab, adding a bullet before the last one, saying something like "Name the library (ex. "RazorUIClassLib"). _(Ensure the name does not end with `.Views` to avoid a namespace collision.)_"
- On the .NET Core CLI tab, adding the same italicized text after the link that provides the link to documentation for `dotnet new`.

OTOH, if the namespace collision shouldn't be an issue (i.e., a bug), then I'd think the doc either wouldn't need to be updated, or could be updated now, and then reverted once the fix is in place.

**Comment by Rick-Anderson** [comment]
@pranavkm can you comment on the suggestion by @danieljsummers ?

**Comment by pranavkm** [comment]
This seems reasonable. Also, it's an issue with file name collision rather than a namespace collision.

**Comment by danieljsummers** [comment]
I'll get it turned tomorrow evening (Sunday) unless something comes up IRL.

**Comment by danieljsummers** [comment]
I ended up moving the bullet up 1 from my original suggestion, because the name of the project precedes the dialog that has the .NET Core 2.1 selection. I also spelled out "filename collision with the generated view library" vs. "namespace collision".
